javaslang/src/main/java/javaslang/collection/Multimaps.java

```diff
@@ -0,0 +1,44 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _______
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  //  /\__\   JΛVΛSLΛNG
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/ \ /__\ \   Copyright 2014-2017 Javaslang, http://javaslang.io
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/\__\_/  \_//  \__/\_____/   Licensed under the Apache License, Version 2.0
+ */
+package javaslang.collection;
+
+import javaslang.Tuple2;
+
+import java.util.Objects;
+import java.util.function.Function;
+
+/**
+ * INTERNAL: Common {@code Multimap} functions (not intended to be public).
+ *
+ * @author Ruslan Sennov, Daniel Dietrich
+ * @since 2.1.0
+ */
+class Multimaps {
+
+    @SuppressWarnings("unchecked")
+    static <K, V, M extends Multimap<K, V>> M ofJavaMap(M map, java.util.Map<? extends K, ? extends V> javaMap) {
+        Objects.requireNonNull(javaMap, "map is null");
```

Comment:
I see that we need _implicitly_ "map is null" for the `javaMap` parameter because of the calling method. It is confusing. Please rename the `javaMap` parameter to `map` and `map` to `source` or `sourceMap`. Then the non-null check makes sense again.